src/lib/netlist/plib/plists.h

```diff
@@ -203,12 +203,14 @@ namespace plib {
 			element_t * m_prev;
 		};
 
-		struct iter_t final : public std::iterator<std::forward_iterator_tag, LC>
+		struct iter_t final
 		{
 		private:
 			element_t * p;
 		public:
 			using tag = std::integral_constant<int, TAG>;
+			using iterator_category = std::forward_iterator_tag;
+			using value_type = LC;
```

Comment:
Do you also need to declare `difference_type`, `reference` and possibly `pointer` for it to be a complete implementation of the iterator concept for C++17?  I think automatic deduction using `std::iter_value_t` and `std::incrementable_traits` are only introduced in C++20.

Reply:
To be honest not sure. Anyway adding next block does not hurt.
```			
			using pointer = value_type *;
			using reference = value_type &;
			using iterator = value_type *;
			using difference_type = std::ptrdiff_t;
```

Reply:
Leave out the `using iterator = ...` otherwise you’ll have a dubious `iterator` type inside the iterator.

Reply:
done